### ISSUE TITLE
[DISCUSSION] railroads

### ISSUE THREAD

**Issue by hrbrmstr** [walkerke/tigris#5]
so, the resultant shapefile from a call to `rails()` results in a **353.3MB** `SpatialLinesDataFrame`. I've got a killer system and even it struggles to do anything useful with it. Since even the d/l take a bit, perhaps we should take advantage of the know user-wait on first grab to split it into chunks based on state? Perhaps even add `state` and `county` params to the function like the others to give the illusion of similar operation to the user. We'd have to add a message saying what's going on to the user for the initial processing and this one would almost require (IMO) mandatory caching vs the optional one now.

**Comment by walkerke** [comment]
Yeah - definitely worth looking into.  I generally still turn to desktop GIS software to handle large geo data like this as R struggles with it.  The catch here is that there isn't a "state" field in the dataset, so to subset we'd have to perform a geometry operation using `gIntersection` or something like that - and I'm not sure how that would go performance-wise.  I'll do a little experimentation with this.

**Comment by walkerke** [comment]
So this works but it is super slow - didn't time it but it took my machine at least 5 minutes.  

```r
library(tigris)
library(rgeos)
library(sp)

railroads <- rails()

all_states <- states()

oregon <- all_states[all_states$NAME == 'Oregon', ]

oregon_rails <- gIntersection(railroads, oregon)

plot(oregon_rails)
```

**Comment by hrbrmstr** [comment]
wow. we may want to consider splitting them all out by states in a separate
`tigrisRails` package. there are data pkg size constraints on CRAN but this
may be allowed given the functionality/speed it'd enable.

**Comment by hrbrmstr** [comment]
Also wondering if this could be sped up with functions from `gdalUtils`. food for thought for my plane ride today.

**Comment by walkerke** [comment]
@hrbrmstr: just wanted to let you know that I'm working on simple features support in tigris in the __sf__ branch (https://github.com/walkerke/tigris/tree/sf).  The (cached) national railroads dataset loads into R on my machine in a couple seconds as simple features.

**Comment by walkerke** [comment]
I've shipped simple features support to CRAN, which handles the railroads much better; I'm closing this issue for now.
